SwiftificationTests/StringTests.swift

```diff
@@ -87,4 +87,21 @@ class StringTests: XCTestCase {
         XCTAssertNil(str[safe: str.startIndex.advancedBy(5, limit: str.endIndex)])
     }
     
-}
\ No newline at end of file
+    func testAppendAndPrependOptionalString() {
+        let str1: String? = "Hi"
+        let str2: String? = " Bob "
+        let strNil: String? = nil
+        
+        let prependString1On2 = str1 + str2 // "Hi Bob "
+        let appendString1On2 = str2 + str1 // " Bob Hi"
+        let onlyString1 = str1 + strNil // "Hi"
+        let onlyString1Append = strNil + str1 // "Hi"
+        
+        XCTAssertEqual(prependString1On2, "Hi Bob ")
+        XCTAssertEqual(appendString1On2, " Bob Hi")
+        XCTAssertEqual(onlyString1, "Hi")
+        XCTAssertEqual(onlyString1Append, "Hi")
```

Comment:
We should also test with two `nil` values. And I think it would be useful to test with an optional string on one side and a non-optional on the other (both combinations), just in case that adds some type weirdness.